modules/gospel.js

```diff
@@ -0,0 +1,315 @@
+    return;
+  }
+  const verseContent = [];
+  for (const num of versesNums) {
+    if (bookJson[bookRef.bookName][chapter][num]) {
+      verseContent.push(num.toString() + " " + bookJson[bookRef.bookName][chapter][num]);
+    }
+  }
+  const verseJoinedContent = verseContent.join("\n\n");
+  if (verses && verseJoinedContent.length === 0) {
+    interaction.reply({ content: "The verse(s) you requested weren't found.", ephemeral: true });
+    return;
+  }
+  embed.setDescription(verseJoinedContent.length > 2048 ? verseJoinedContent.slice(0, 2048) + "…" : verseJoinedContent);
+  embed.setColor(0x012b57);
+  interaction.reply({ embeds: [embed] });
+}
+
+/**
+ * Splits the verses section in a scripture reference into individual verse numbers.
+ * @param {string} verses A string containing numbers, spaces, hyphens, and commas.
+ * @returns An Array of numbered integers as interpreted. "3-5, 7" returns [3, 4, 5, 7]
+ */
+function parseVerseRange(verses) {
+  let versesNums;
+  if (verses) {
+    verses = verses.replace(/ /g, "");
+    const versesList = verses.split(",");
+    versesNums = new Array();
+    const rangeRegex = /(\d+)(?:-(\d+))?/;
+    for (const range of versesList) {
+      const results = range.match(rangeRegex);
+      const low = results[1],
+        high = results[2];
+      if (!low) {
+        throw new SyntaxError("Invalid verse range.");
+      } else if (!high) {
+        versesNums.push(parseInt(low));
+      } else {
+        let lowNum = parseInt(low);
+        let highNum = parseInt(high);
+        // Swap the range if it's out of order.
+        if (lowNum > highNum) {
+          [lowNum, highNum] = [highNum, lowNum];
+        }
+        for (let i = lowNum; i <= highNum; i++) {
+          versesNums.push(i);
+        }
+      }
+    }
+    // Get unique verses
+    versesNums = [...new Set(versesNums)].sort((a, b) => a - b);
+  } else { versesNums = []; }
+  return versesNums;
+}
+
+async function slashGospelComeFollowMe(interaction) {
+  // Most of this function is old code. Not sure how to improve it.
+  let date = new Date();
+  date.setHours(0, 0, 0, 0);
+  const displayDate = new Date(date);
+  let jan1;
+  // Account for year-end dates.
+  if (date.getMonth() == 11 && (date.getDate() - date.getDay() >= 26)) {
+    jan1 = new Date(date.getFullYear() + 1, 0, 1, 0, 0, 0);
+    date = jan1;
+  } else {
+    jan1 = new Date(date.getFullYear(), 0, 1, 0, 0, 0);
+  }
+
+  const manual = manuals.get(date.getFullYear());
+  if (manual) {
+    // Add full weeks and check partial weeks by day of week comparison
+    let week = ((date.getDay() + 6) % 7 < (jan1.getDay() + 6) % 7 ? 2 : 1) + Math.floor((date - jan1) / (1000 * 60 * 60 * 24 * 7));
+    // Account for General Conference
+    if ((date.getMonth() == 3 && (date.getDate() - date.getDay()) >= 0) || date.getMonth() > 3) week -= 1;
+    if ((date.getMonth() == 9 && (date.getDate() - date.getDay()) >= 0) || date.getMonth() > 9) week -= 1;
```

Comment:
This was necessary in 2020, but now results in the wrong week. These lines can be commented out (in case they'll be needed in future years).